types/tableau-js-api/index.d.ts

```diff
@@ -0,0 +1,29 @@
+// Type definitions for Tableau Javascript API 2.8
+// Project: https://github.com/tableau-developers/ts-types-tableau-js-api, https://help.tableau.com/current/api/js_api/en-us/JavaScriptAPI/js_api.htm
+// Definitions by: John Hegele <https://github.com/jhegele>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/**
+ * Type definitions were built using Tableau's API documentation as a reference:
+ *
+ * https://help.tableau.com/v2021.1/api/js_api/en-us/JavaScriptAPI/js_api_ref.htm
+ *
+ * Where possible, the structure of the type definition mirros what is present
+ * the documentation. The file structure also attempts to (largely) mirror the
+ * structure of the documentation. The Viz Classes category is located in the
+ * viz.d.ts file and the Sheet Classes category is located in the sheet.d.ts
+ * file, e.g.
+ *
+ * For ease of managing and updating these defintions in the future, all enums
+ * have been split out into enums.d.ts.
+ *
+ * There are some undocumented classes and methods in Tableau's Javascript API
+ * (tableau.Version(), e.g.). Because those classes and methods are not
+ * included in Tableau's public documentation, they have been intentionally
+ * excluded here.
+ */
+
+export as namespace tableau;
```

Comment:
This is inaccurate since it sounds like this package is for browser-only code, not node modules.
On the other hand, it's MUCH more convenient and there's no module to get wrong, so I think it's fine.

Reply:
@sandersn can you give me some detail on how this should be structured? I'm OK to leave it as-is but, since I'm still learning a lot about writing type defs, it would be helpful for me to better understand why this is technically incorrect even if it does work. Thanks!

Reply:
Thanks, very informative! I actually started building this using `namespace` but I ran into some challenges getting things to export properly when I tested it in my own code. Like you, I'm much more comfortable and familiar with the `import/export` syntax so that's what I defaulted to. Greatly appreciate the explanation. Really helps to improve my understanding, conceptually, of how to think about type defs going forward.